src/keypair.rs

```diff
@@ -0,0 +1,183 @@
+    pub fn new_from_private_key(private_key: &[u8]) -> Keypair {
+        let secret_key = SecretKey::from_slice(private_key).expect("32 bytes, within curve order");
+
+        return Keypair {
+            private_key: secret_key,
+        };
+    }
+    /// Create and return a keypair with  the given hex u8 array as the private key
+    pub fn new_from_private_key_hex(private_key_hex: &str) -> Keypair {
+        //let mock_private_key = "da79fe6d86347e8f8dc71eb3dbab9ba5623eaaed6c5dd0bb257c0d631faaff16";
+        let mut bytes = [0u8; 32];
+        hex::decode_to_slice(private_key_hex, &mut bytes as &mut [u8]);
+        return Keypair::new_from_private_key(&bytes);
+    }
+    /// Get the public key of the keypair in base58(i.e. address) format
+    pub fn get_address(&self) -> String {
+        return PublicKey::from_secret_key(&SECP, &self.private_key)
+            .serialize()
+            .to_base58();
+    }
+    /// Get the public key of the keypair as secp256k1::key::PublicKey
+    pub fn get_public_key(&self) -> PublicKey {
+        return PublicKey::from_secret_key(&SECP, &self.private_key);
+    }
+    /// Get the private key as a hex-encoded string
+    pub fn get_private_key(&self) -> String {
+        return self.private_key.to_string();
+    }
+    /// Hash the message string with sha256 for signing by secp256k1 and return as byte array
+    /// TODO: Make sure this handles utf correctly. We probably want to ensure that the message
+    /// is actually just ascii encoded...
+    fn make_message_from_string(message_string: &str) -> [u8; 32] {
+        let mut hasher = Sha256::new();
+        hasher.update(message_string.as_bytes());
+        let hashvalue = hasher.finalize();
+        return [
+            hashvalue[0],
+            hashvalue[1],
+            hashvalue[2],
+            hashvalue[3],
+            hashvalue[4],
+            hashvalue[5],
+            hashvalue[6],
+            hashvalue[7],
+            hashvalue[8],
+            hashvalue[9],
+            hashvalue[10],
+            hashvalue[11],
+            hashvalue[12],
+            hashvalue[13],
+            hashvalue[14],
+            hashvalue[15],
+            hashvalue[16],
+            hashvalue[17],
+            hashvalue[18],
+            hashvalue[19],
+            hashvalue[20],
+            hashvalue[21],
+            hashvalue[22],
+            hashvalue[23],
+            hashvalue[24],
+            hashvalue[25],
+            hashvalue[26],
+            hashvalue[27],
+            hashvalue[28],
+            hashvalue[29],
+            hashvalue[30],
+            hashvalue[31],
+        ];
+    }
+    /// Hash and sign a string
+    pub fn sign_string_message(&self, message_string: &str) -> String {
+        let message_bytes = Keypair::make_message_from_string(message_string);
+        let bytes = self.sign_message(&message_bytes);
+        let mut string_out = String::new();
+        write!(&mut string_out, "{:?}", bytes);
```

Comment:
We changed the method to return a Result... but maybe it would be okay to just expect()... I think I prefer to return a Result but that may be getting too academic, if write! cannot convert bytes to a string perhaps there are bigger problems than behavior or saito-rust...